### PR TITLE
Controllers mapping created from JSON, first draft

### PR DESCRIPTION
**Pull request by samcake**
- creating the mapping definition from a JSON string

- add the missing functions in UserInputMapper to find an INput from a string "deviceName.inputChannel" 

- added a"baseName" to the DeviceProxy to simplify the naming scheme maybe?

- a Factory for Controller::Filter so we can create the FIlter based on their "type" from the JSON and then parse the missing parameters

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 12e103c90c03c707c9c13601166e06ee99d17fb2 available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

File: libraries/controllers/src/controllers/impl/RouteBuilderProxy.cpp [hunk 1]
```diff
@@ -54,84 +57,41 @@ QObject* RouteBuilderProxy::filter(const QScriptValue& expression) {
 
 
 QObject* RouteBuilderProxy::clamp(float min, float max) {
-    addFilter([=](float value) {
-        return glm::clamp(value, min, max);
-    });
+    addFilter(Filter::Pointer(new ClampFilter(min, max)));
```
Comment (jherico) [inline libraries/controllers/src/controllers/impl/RouteBuilderProxy.cpp]: I'm not sure if it's in the coding standard, but the preferred way to do this is `std::make_shared<ClampFilter>(min, max)`

**Comment by hifi-gustavo** [comment]
Build console output for commit: cdfe9788d13d6fdf7ccdf58ca50621323b7a9f7f available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 416df1c44c7f1c181ab7fd81e66f5888a3bfb4a2 available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ad6c2067b08261bfe0bce14ccd28259c49b0054 available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 249efa383ea41d534d802426fb966a4a5b17cfe9 available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1b03b6867c075a1c4b4430a0a682a8a4d9fb62aa available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 63ad9ae19837c2d829a42a63cb99a108d82350d8 available at: https://gist.github.com/f149fb63b18fc5bb54d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6081.dmg

**Pull request merged**